editor/components/block-list/block.js

```diff
@@ -428,6 +434,43 @@ export class BlockListBlock extends Component {
 		this.setState( { error } );
 	}
 
+	onDragStart( event ) {
+		this.props.onDragStart(
+			event,
+			this.props.uid,
+			`block-${ this.props.uid }`,
+			this.props.order,
+			BLOCK_REORDER
+		);
+		// Following fires after the events in props.onDragStart.
+		setTimeout( () => {
+			this.setState( { dragging: true } );
+		}, 0 );
+	}
+
+	onDragEnd( event ) {
+		this.props.onDragEnd( event );
+		// Following fires after the events in props.onDragStart.
+		setTimeout( () => {
+			this.setState( { dragging: false } );
+		}, 0 );
+	}
+
+	/*
+	 * Reorder via Drag & Drop. Final step.
+	 * Strategy:
+	 *  - Call dragEnd handler.
+	 *    - We call the dragEnd handler here to ensure the dropzone does not prevent this call.
+	 *  - Initiate reordering.
+	 */
+	reorderBlock( event, uid, toIndex ) {
+		this.onDragEnd( event );
+		// Following fires after the events in props.onDragEnd.
+		setTimeout( () => {
```

Comment:
Why do we need to wait here?

Reply:
See my comment above. I will aim to get rid of these.